lib/model/rwfolder.go

```diff
@@ -780,13 +778,9 @@ func (f *rwFolder) renameFile(source, target protocol.FileInfo) {
 	from := filepath.Join(f.dir, source.Name)
 	to := filepath.Join(f.dir, target.Name)
 
-	if f.versioner != nil {
```

Comment:
I'm a little bit confused about the direction of this review ... when I read it now, there is no copy-delete in the default case. The `default` versioner does the 'else branch' `err = osutil.TryRename(from, to)` and the other versioners do the 'if branch'

```go
err = osutil.Copy(from, to)		
 		if err == nil {		
 			err = osutil.InWritableDir(f.versioner.Archive, from)		
 		}
```